stdlib/asyncio/taskgroups.pyi

```diff
@@ -18,3 +19,7 @@ class TaskGroup:
     async def __aenter__(self) -> Self: ...
     async def __aexit__(self, et: type[BaseException] | None, exc: BaseException | None, tb: TracebackType | None) -> None: ...
     def create_task(self, coro: _CoroutineLike[_T], *, name: str | None = None, context: Context | None = None) -> Task[_T]: ...
+    def _on_task_done(self, task: Task[Any]) -> None: ...
+
+    _loop: AbstractEventLoop | None
```

Comment:
Style nit: please, add attributes first (before all methods)

Reply:
I made the change. However, I have a question about runtests.py. When I ran it I got a bunch of errors that seem to be unrelated to the changes I made. Is there some change I need to make to my development environment?

I ran `%> python ./scripts/runtests.py --run-stubtest stdlib/asyncio`

and got the output

```shell
Running ruff...

Running isort...

Running Black...
All done! ✨ 🍰 ✨
32 files left unchanged.

Running Flake8...

Running check_consistent.py...

Running check_new_syntax.py...

Running Pyright (stricter configs) for Python 3.8...
9.8.0
Running: /opt/homebrew/bin/npx pyright@1.1.323 stdlib/asyncio --pythonversion 3.8 -p pyrightconfig.stricter.json
0 errors, 0 warnings, 0 informations


Running mypy for Python 3.8...
*** Testing Python 3.8 on darwin
Testing stdlib (32 files)...success

--- success, 32 files checked ---

Running stubtest...
{redacted}/typeshed/.venv/bin/python -m mypy.stubtest --check-typeshed --custom-typeshed-dir . --allowlist tests/stubtest_allowlists/py3_common.txt --allowlist tests/stubtest_allowlists/py311.txt --allowlist tests/stubtest_allowlists/darwin.txt --allowlist tests/stubtest_allowlists/darwin-py311.txt
error: _curses.BUTTON5_CLICKED is not present in stub
Stub: in file stdlib/_curses.pyi
MISSING
Runtime:
4194304

error: _curses.BUTTON5_DOUBLE_CLICKED is not present in stub
Stub: in file stdlib/_curses.pyi
MISSING
Runtime:
8388608

error: _curses.BUTTON5_PRESSED is not present in stub
Stub: in file stdlib/_curses.pyi
MISSING
Runtime:
2097152

error: _curses.BUTTON5_RELEASED is not present in stub
Stub: in file stdlib/_curses.pyi
MISSING
Runtime:
1048576

error: _curses.BUTTON5_TRIPLE_CLICKED is not present in stub
Stub: in file stdlib/_curses.pyi
MISSING
Runtime:
16777216

error: builtins.input is inconsistent, runtime argument "prompt" has a default value of None, which is different from stub argument default ''
Stub: in file stdlib/builtins.pyi:1447
def (builtins.object =) -> builtins.str
Runtime:
def (prompt=None, /)

error: curses.BUTTON5_CLICKED is not present in stub
Stub: in file stdlib/curses/__init__.pyi
MISSING
Runtime:
4194304

error: curses.BUTTON5_DOUBLE_CLICKED is not present in stub
Stub: in file stdlib/curses/__init__.pyi
MISSING
Runtime:
8388608

error: curses.BUTTON5_PRESSED is not present in stub
Stub: in file stdlib/curses/__init__.pyi
MISSING
Runtime:
2097152

error: curses.BUTTON5_RELEASED is not present in stub
Stub: in file stdlib/curses/__init__.pyi
MISSING
Runtime:
1048576

error: curses.BUTTON5_TRIPLE_CLICKED is not present in stub
Stub: in file stdlib/curses/__init__.pyi
MISSING
Runtime:
16777216

error: platform.uname_result._fields variable differs from runtime type tuple[Literal['system'], Literal['node'], Literal['release'], Literal['version'], Literal['machine']]
Stub: in file stdlib/platform.pyi:-1
tuple[builtins.str, builtins.str, builtins.str, builtins.str, builtins.str, builtins.str]
Runtime:
('system', 'node', 'release', 'version', 'machine')

error: ssl.OP_IGNORE_UNEXPECTED_EOF is not present in stub
Stub: in file stdlib/ssl.pyi
MISSING
Runtime:
<Options.OP_IGNORE_UNEXPECTED_EOF: 128>

error: ssl.Options.OP_IGNORE_UNEXPECTED_EOF is not present in stub
Stub: in file stdlib/ssl.pyi
MISSING
Runtime:
<Options.OP_IGNORE_UNEXPECTED_EOF: 128>

error: unittest.mock._patch_dict.decorate_async_callable is not present at runtime
Stub: in file stdlib/unittest/mock.pyi:286
def [_AF <: def (*Any, **Any) -> typing.Coroutine[Any, Any, Any]] (self: unittest.mock._patch_dict, f: _AF`-1) -> _AF`-1
Runtime:
MISSING

error: unittest.mock._patch_dict.decorate_callable is not present at runtime
Stub: in file stdlib/unittest/mock.pyi:285
def [_F <: def (*Any, **Any) -> Any] (self: unittest.mock._patch_dict, f: _F`-1) -> _F`-1
Runtime:
MISSING

note: unused allowlist entry enum.auto.__init__
note: unused allowlist entry enum.auto.value
note: unused allowlist entry unittest.TestCase.__init_subclass__
note: unused allowlist entry unittest.case.TestCase.__init_subclass__
note: unused allowlist entry _?curses.A_ITALIC
Found 21 errors (checked 541 modules)

NB: stubtest output depends on the Python version (and system) it is run with. See README.md for more details.
NB: We only check positional-only arg accuracy for Python 3.10.

{redacted}/typeshed/.venv/bin/python -m mypy.stubtest --check-typeshed --custom-typeshed-dir . --allowlist tests/stubtest_allowlists/py3_common.txt --allowlist tests/stubtest_allowlists/py311.txt --allowlist tests/stubtest_allowlists/darwin.txt --allowlist tests/stubtest_allowlists/darwin-py311.txt




To fix "unused allowlist" errors, remove the corresponding entries from tests/stubtest_allowlists

Running pytype...
pytype does not support Python 3.11+ yet.

Running Pyright regression tests for Python 3.8...
9.8.0
Running: /opt/homebrew/bin/npx pyright@1.1.323 test_cases --pythonversion 3.8 -p pyrightconfig.testcases.json
0 errors, 0 warnings, 0 informations


Running mypy regression tests for Python 3.8...
Running mypy --platform darwin --python-version 3.8 on the standard library test cases... success

Test completed successfully!



--- TEST SUMMARY: One or more tests failed. See above for details. ---

Flake8: Success
Check consistent: Success
Check new syntax: Success
Pyright: Success
mypy: Success
stubtest: Failed
pytype: Failed
Pyright regression tests: Success
mypy regression test: Success
```

Reply:
> However, I have a question about runtests.py. When I ran it I got a bunch of errors that seem to be unrelated to the changes I made.

What's the output you get from running `python -VV` on the command line?

Reply:
`Python 3.11.0 | packaged by conda-forge | (main, Oct 25 2022, 06:21:25) [Clang 14.0.4 ]`

Reply:
A bunch of your stubtest errors will probably go away if you upgrade to 3.11.4. In general, we only test stubtest in CI using the latest micro version of each feature release of Python. (We could probably document this better.)

There may also be a few differences between conda-packaged Python and the Python builds we use in CI. Again, we could probably document this better :/

Reply:
I made a new venv starting with python installed with homebrew. Here is the output of `python -VV` in the new venv.
`Python 3.11.5 (main, Aug 24 2023, 15:09:45) [Clang 14.0.3 (clang-1403.0.22.14.1)]`

I installed the test requirements and ran runtests.py again. I got a huge number of errors, mostly dealing with tkinter. Installing the brew package `python-tk` solved most of them but I am left with output below. These errors are different from the ones generated with the conda distribution of 3.11.0 

```sh

Running ruff...

Running isort...

Running Black...
All done! ✨ 🍰 ✨
32 files left unchanged.

Running Flake8...

Running check_consistent.py...

Running check_new_syntax.py...

Running Pyright (stricter configs) for Python 3.8...
9.8.0
Running: /opt/homebrew/bin/npx pyright@1.1.323 stdlib/asyncio --pythonversion 3.8 -p pyrightconfig.stricter.json
0 errors, 0 warnings, 0 informations


Running mypy for Python 3.8...
*** Testing Python 3.8 on darwin
Testing stdlib (32 files)...success

--- success, 32 files checked ---

Running stubtest...
{redacted}/typeshed/.venv/bin/python -m mypy.stubtest --check-typeshed --custom-typeshed-dir . --allowlist tests/stubtest_allowlists/py3_common.txt --allowlist tests/stubtest_allowlists/py311.txt --allowlist tests/stubtest_allowlists/darwin.txt --allowlist tests/stubtest_allowlists/darwin-py311.txt
error: enum.__all__ names exported from the stub do not correspond to the names exported at runtime. This is probably due to things being missing from the stub or an inaccurate `__all__` in the stub
Stub: in file stdlib/enum.pyi
Names exported in the stub but not at runtime: []
Runtime:
Names exported at runtime but not in the stub: ['pickle_by_enum_name', 'pickle_by_global_name']

error: enum.Enum.__copy__ is not present in stub
Stub: in file stdlib/enum.pyi
MISSING
Runtime: in file /opt/homebrew/Cellar/python@3.11/3.11.5/Frameworks/Python.framework/Versions/3.11/lib/python3.11/enum.py:1233
<function Enum.__copy__ at 0x104fe8720>

error: enum.Enum.__deepcopy__ is not present in stub
Stub: in file stdlib/enum.pyi
MISSING
Runtime: in file /opt/homebrew/Cellar/python@3.11/3.11.5/Frameworks/Python.framework/Versions/3.11/lib/python3.11/enum.py:1230
<function Enum.__deepcopy__ at 0x104fe8680>

error: enum.pickle_by_enum_name is not present in stub
Stub: in file stdlib/enum.pyi
MISSING
Runtime: in file /opt/homebrew/Cellar/python@3.11/3.11.5/Frameworks/Python.framework/Versions/3.11/lib/python3.11/enum.py:1305
<function pickle_by_enum_name at 0x104fe8c20>

error: enum.pickle_by_global_name is not present in stub
Stub: in file stdlib/enum.pyi
MISSING
Runtime: in file /opt/homebrew/Cellar/python@3.11/3.11.5/Frameworks/Python.framework/Versions/3.11/lib/python3.11/enum.py:1300
<function pickle_by_global_name at 0x104fe2980>

error: os.lseek is inconsistent, stub argument "__how" differs from runtime argument "whence"
Stub: in file stdlib/os/__init__.pyi:605
def (builtins.int, builtins.int, builtins.int) -> builtins.int
Runtime:
def (fd, position, whence, /)

error: posix.lseek is inconsistent, stub argument "__how" differs from runtime argument "whence"
Stub: in file stdlib/posix.pyi:605
def (builtins.int, builtins.int, builtins.int) -> builtins.int
Runtime:
def (fd, position, whence, /)

error: pydoc.Helper.help is inconsistent, stub does not have argument "is_cli"
Stub: in file stdlib/pydoc.pyi:236
def (self: pydoc.Helper, request: Any)
Runtime: in file /opt/homebrew/Cellar/python@3.11/3.11.5/Frameworks/Python.framework/Versions/3.11/lib/python3.11/pydoc.py:2054
def (self, request, is_cli=False)

error: pydoc.doc is inconsistent, stub does not have argument "is_cli"
Stub: in file stdlib/pydoc.pyi:211
def (thing: Union[builtins.str, builtins.object], title: builtins.str =, forceload: builtins.bool =, output: Union[_typeshed.SupportsWrite[builtins.str], None] =)
Runtime: in file /opt/homebrew/Cellar/python@3.11/3.11.5/Frameworks/Python.framework/Versions/3.11/lib/python3.11/pydoc.py:1780
def (thing, title='Python Library Documentation: %s', forceload=0, output=None, is_cli=False)

error: ssl.OP_IGNORE_UNEXPECTED_EOF is not present in stub
Stub: in file stdlib/ssl.pyi
MISSING
Runtime:
<Options.OP_IGNORE_UNEXPECTED_EOF: 128>

error: ssl.Options.OP_IGNORE_UNEXPECTED_EOF is not present in stub
Stub: in file stdlib/ssl.pyi
MISSING
Runtime:
<Options.OP_IGNORE_UNEXPECTED_EOF: 128>

note: unused allowlist entry types.CodeType.replace
Found 12 errors (checked 541 modules)

NB: stubtest output depends on the Python version (and system) it is run with. See README.md for more details.
NB: We only check positional-only arg accuracy for Python 3.10.

Command run was: {redacted}/typeshed/.venv/bin/python -m mypy.stubtest --check-typeshed --custom-typeshed-dir . --allowlist tests/stubtest_allowlists/py3_common.txt --allowlist tests/stubtest_allowlists/py311.txt --allowlist tests/stubtest_allowlists/darwin.txt --allowlist tests/stubtest_allowlists/darwin-py311.txt




To fix "unused allowlist" errors, remove the corresponding entries from tests/stubtest_allowlists

Running pytype...
pytype does not support Python 3.11+ yet.

Running Pyright regression tests for Python 3.8...
9.8.0
Running: /opt/homebrew/bin/npx pyright@1.1.323 test_cases --pythonversion 3.8 -p pyrightconfig.testcases.json
0 errors, 0 warnings, 0 informations


Running mypy regression tests for Python 3.8...
Running mypy --platform darwin --python-version 3.8 on the standard library test cases... success

Test completed successfully!



--- TEST SUMMARY: One or more tests failed. See above for details. ---

Flake8: Success
Check consistent: Success
Check new syntax: Success
Pyright: Success
mypy: Success
stubtest: Failed
pytype: Failed
Pyright regression tests: Success
mypy regression test: Success
```

Reply:
> I made a new venv starting with python installed with homebrew. Here is the output of `python -VV` in the new venv.
> `Python 3.11.5 (main, Aug 24 2023, 15:09:45) [Clang 14.0.3 (clang-1403.0.22.14.1)]`

Python 3.11.5 hasn't made its way to GitHub Actions yet, unfortunately -- we're still using Python 3.11.4 in CI. Because there are often minor API changes with each micro Python release, we can only guarantee that stubtest will pass locally if you're using _exactly_ the same Python version as we're using in CI. We'll likely have to make one or two changes to fix our CI when Python 3.11.5 becomes available for use in GitHub Actions.